spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/query/QueryValidator.java

```diff
@@ -0,0 +1,116 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.spi.query;
+
+import org.eclipse.dataspaceconnector.common.reflection.ReflectionUtil;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.ParameterizedType;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+/**
+ * Validates that a particular query is valid, i.e. contains a left-hand operand, that conforms to the canonical
+ * format.
+ */
+public class QueryValidator {
```

Comment:
I would add some unit tests here as well, to verify all the conditions (invalid path, query map, field not found...)